File: Phan_A/baitap32/baitap32.py
import sys

def is_square(n):
    if n == 0 or n == 1:
        return True
    
    left = 1
    right = min(n, 10**9)
    
    while left <= right:
        mid = (left + right) // 2
        sq = mid * mid
        
        if sq == n:
            return True
        elif sq < n:
            left = mid + 1
        else:
            right = mid - 1
            
    return False

def main():
    input_data = sys.stdin.read().split()
    if not input_data:
        return
    n = int(input_data[0])
    if is_square(n):
        print("YES")
    else:
        print("NO")

if __name__ == "__main__":
    main()

File: Phan_A/baitap32/baitap32.jl
function is_square(n::UInt128)
    if n == 0 || n == 1
        return true
    end

    left = UInt128(1)
    right = min(n, UInt128(10^9))

    while left <= right
        mid = left + div(right - left, 2)
        sq = mid * mid

        if sq == n
            return true
        elseif sq < n
            left = mid + 1
        else
            right = mid - 1
        end
    end
    return false
end

function main()
    input_data = split(read(stdin, String))
    if isempty(input_data)
        return
    end

    n = parse(UInt128, input_data[1])
    if is_square(n)
        println("YES")
    else
        println("NO")
    end
end

main()